### ISSUE TITLE
Angular 2.0.1 breaks p-dataTable expandableRows

### ISSUE THREAD

**Issue by ribizli** [primefaces/primeng#966]
The following change breaks how p-dataTable queries for the expandableRow template:
core: ContentChild descendants should be queried by default ([0dc15eb](https://github.com/angular/angular/commit/0dc15eb)), closes [#11645](https://github.com/angular/angular/issues/11645).

The issue in my case was that an other deeper located template (used for a column body) was taken for the row-expansion, but the let-[variable] was not set to the `$implicit` value (actually was set to `"rowData"` which is `undefined` in this case).

Please introduce a selector directive (e.g. `pRowExpansion`, like `pTemplate` for p-column) for row-expansion, and use as selector for `@ContentChild(PrimeRowExpansion)`.

**Comment by cagataycivici** [comment]
I couldn't replicate on 2.0.1, can you please post your p-dataTable code.

**Comment by ribizli** [comment]
Just replace a column in your example whit a template:
```
  <p-column field="vin" header="Vin">
      <template pTemplate let-x="rowData">{{x.vin}}</template>
  </p-column>
```

**Comment by ribizli** [comment]
I've provided you with a solution PR, you rejected it, that you'll do it your way. And now you released the new beta version without fixing this breaking problem.

Why is this project open source, if you don't expect contributors?

**Comment by ilianiv** [comment]
@cagataycivici, 
```
@ContentChild(TemplateRef) rowExpansionTemplate: TemplateRef<any>; 
```
Finds a p-column's template instead of p-dataTables's one because it queries it's descendants by default since 2.0.1.

Moving p-dataTable's template tag above p-columns fixes the issue.

You can replicate it with your own example by adding p-column's template:
```
<p-dataTable [value]="cars" expandableRows="true">
    <p-column expander="true" styleClass="col-icon"></p-column>
    <p-column header="Vin">
      <template let-col let-car="rowData" pTemplate type="body">
        <span>{{car.vin}}</span>
      </template>
    </p-column>
    <p-column field="year" header="Year"></p-column>
    <p-column field="brand" header="Brand"></p-column>
    <p-column field="color" header="Color"></p-column>
    <template let-car>
        <div class="ui-grid ui-grid-responsive ui-fluid" style="font-size:16px;padding:20px">
            <div class="ui-grid-row">
                <div class="ui-grid-col-3" style="text-align:center">
                    <i class="fa fa-search" (click)="showCar(car)" style="cursor:pointer;float:left;margin-top:40px"></i>
                    <img src="showcase/resources/demo/images/car/{{car.brand}}-big.gif">
                </div>
                <div class="ui-grid-col-9">
                    <div class="ui-grid ui-grid-responsive ui-grid-pad">
                        <div class="ui-grid-row">
                            <div class="ui-grid-col-2 label">Vin: </div>
                            <div class="ui-grid-col-10">{{car.vin}}</div>
                        </div>
                        <div class="ui-grid-row">
                            <div class="ui-grid-col-2 label">Year: </div>
                            <div class="ui-grid-col-10">{{car.year}}</div>
                        </div>
                        <div class="ui-grid-row">
                            <div class="ui-grid-col-2 label">Brand: </div>
                            <div class="ui-grid-col-10">{{car.brand}}</div>
                        </div>
                        <div class="ui-grid-row">
                            <div class="ui-grid-col-2 label">Color: </div>
                            <div class="ui-grid-col-10">{{car.color}}</div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </template>
</p-dataTable>
```

**Comment by ribizli** [comment]
@ilianiv thanks for your workaround

**Comment by cagataycivici** [comment]
Fixed via #1396
